src/newshop.cpp

```diff
@@ -374,11 +391,22 @@ bool install_ware_in_target_character(struct obj_data *ware, struct char_data *i
     return FALSE;
   }
 
+  // Double Bioindex Loss for Drakes.
+  if (GET_OBJ_TYPE(ware) == ITEM_BIOWARE) {
+    int biocost = GET_BIOWARE_ESSENCE_COST(ware);
+    if (GET_RACE(recipient) >= RACE_DRAKE_HUMAN && GET_RACE(recipient) <= RACE_DRAKE_TROLL)
+      biocost = GET_BIOWARE_ESSENCE_COST(ware) *= 2;
```

Comment:
I think the use of `*= 2` here is going to cause you issues down the line. What this code does is it doubles _and saves_ the essence cost of the 'ware, which means the object itself now reflects a doubled essence cost. I'm betting that if you buy 'ware and then immediately sell it, you'll pay the doubled cost and then have a 4x essence hole. Meanwhile, if you quit and reload first (causing the 'ware's cost to be loaded from default values), you'll have the expected 2x essence hole.

Recommend using `biocost *= 2;` to reflect that you're doubling the biocost.

Reply:
Updated too prescribed changed, should be working fine now.

Reply:
I'd remove this whole added block since you've written similar code at line 537.